### DISCUSSION TITLE
Loop back the video after it ends

### DISCUSSION THREAD

**Discussion by avidseeker** [mpv-player/mpv#15600] [Q&A]
This is similar to how Youtube player behaves.

- Watch a YT video to the end.
- It stops.
- Pressing space replays the video from the beginning.

Note how pressing space has a different behavior depending on whether the video ended or not.

Is this possible in mpv?

**Comment by Keith94** [comment]
You have to use a .lua script. Here's an example from https://github.com/Samillion/ModernZ/issues/291#issuecomment-2571817344

```lua
mp.observe_property("eof-reached", "bool", function(_, val)
    if val then
        mp.set_key_bindings({
           {"SPACE", function() mp.command("seek 0 absolute; cycle pause") end},
           {"MBTN_LEFT", function() mp.command("seek 0 absolute; cycle pause") end},
        }, "my-playback-restart", "force")
        
        mp.enable_key_bindings("my-playback-restart")
    else
        mp.disable_key_bindings("my-playback-restart")
    end
end)
```
